File: easy/2315_count_asterisks/solution.py
class Solution:
    def countAsterisks(self, s: str) -> int:
        asterisks_count = 0
        in_pair = False

        for c in s:
            if c == "|":
                in_pair = not in_pair
            elif not in_pair and c == "*":
                asterisks_count += 1

        return asterisks_count

File: easy/2315_count_asterisks/solution.cpp
#include <string>

class Solution {
public:
  int countAsterisks(std::string s) {
    int asterisksCount = 0;
    bool inPair = false;

    for (const char &c : s) {
      if (c == '|')
        inPair = !inPair;
      else if (!inPair && c == '*')
        ++asterisksCount;
    }

    return asterisksCount;
  }
};
